### PR TITLE
Add help command

### LINKED ISSUE #66: Add help command
We should have a command which lists bot commands for user friendliness

### LINKED ISSUE #70: Use parser for chat bot
The chat bot currently uses a series of if statements to match a command, but this is error prone (for example 'release notes' was treated as a 'release' command before I switched the order of the if expressions). There are simple parsers available which should let us format the commands like 'release %word', making the code more readable and reliable

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #66 
Fixes #70

#### What's this PR do?
Adds a help command, better parsing of syntax errors

#### How should this be manually tested?
Say `@doof help` and see the list of commands. Also try to start a release for a version number which has an invalid syntax, eg `a.b.c`

### CODE REVIEW

**Comment by annagav** [comment]
This is what I got:
screen shot 2017-12-22 at 2 21 39 pm

**Comment by noisecapella** [comment]
Yes that looks right, except for the `Invalid command format` below. I'll investigate

**Comment by noisecapella** [comment]
@rhysyngsun would you mind taking a look at this when you get a chance? @annagav looked at it already but I made a few cosmetic changes after

File: bot.py [hunk 1]
```diff
@@ -432,33 +569,52 @@ def translate_slack_usernames(self, unchecked_authors):
...
+                                word=arg,
+                            )
+                        )
+                        return
+
+                await command_func(
+                    CommandArgs(
+                        repo_info=repo_info,
+                        channel_id=channel_id,
+                        args=[parser(arg) for arg, parser in zip(args, parsers)],
```
Comment (rhysyngsun) [inline bot.py]: Should this just be passing `parsed_args` instead? If not, it doesn't appear we use `parsed_args` anywhere.

Comment (noisecapella) [reply bot.py]: Yes, I just fixed it

**Pull request merged**